clients/dataset/dataset.go

```diff
@@ -136,6 +139,11 @@ func (c *Client) GetByPath(ctx context.Context, path string) (m Model, err error
 
 	req = setCollectionID(ctx, req)
 
+	if len(authToken) > 0 {
```

Comment:
Like I said above you are having to repeat this multiple times in this method - feels like it makes sense to pull it out into its own thing.

Reply:
simplified as per slack chat.